src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -555,20 +558,36 @@ public void cleanupCourse(Long courseId) {
     public ResponseEntity<List<User>> getAllUsersInGroup(Course course, String groupName) {
         authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
         var usersInGroup = userRepository.findAllInGroup(groupName);
-        usersInGroup.forEach(user -> {
-            // remove some values which are not needed in the client
-            user.setLastNotificationRead(null);
-            user.setActivationKey(null);
-            user.setLangKey(null);
-            user.setLastNotificationRead(null);
-            user.setLastModifiedBy(null);
-            user.setLastModifiedDate(null);
-            user.setCreatedBy(null);
-            user.setCreatedDate(null);
-        });
+        removeUserVariables(usersInGroup);
         return ResponseEntity.ok().body(usersInGroup);
     }
 
+    /**
+     * Search for users of all user groups by login or name in course
+     *
+     * @param course        Course in which to search students
+     * @param nameOfUser    Login or name by which to search students
+     * @return users whose login matched
+     */
+    public List<User> searchOtherUsersNameInCourse(Course course, String nameOfUser) {
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
```

Comment:
Why do you perform this check here and not at the corresponding Resource? In fact, no user is even passed (null) so can this even fail?

Reply:
@ole-ve The check works even if the user is passed null. Please see the attachment.
![image](https://user-images.githubusercontent.com/67344158/163027176-f5e98a95-fc79-45a0-8ada-c72e42fe3d6c.png)